db.py

```diff
@@ -0,0 +1,111 @@
+"""
+This module handles the management of the database.
+"""
+import enum
+import os
+from typing import List
+
+import firebase_admin
+from firebase_admin import credentials
+from firebase_admin import firestore
+from google.cloud.firestore_v1 import base_query
+import gradio as gr
+
+from credentials import get_credentials_json
+
+
+def get_environment_variable(name: str) -> str:
+  value = os.getenv(name)
+  if value is None:
+    raise ValueError(f"Environment variable {name} is not set")
+  return value
+
+
+RATINGS_COLLECTION = get_environment_variable("RATINGS_COLLECTION")
+SUMMARIZATIONS_COLLECTION = get_environment_variable(
+    "SUMMARIZATIONS_COLLECTION")
+TRANSLATIONS_COLLECTION = get_environment_variable("TRANSLATIONS_COLLECTION")
+
+if gr.NO_RELOAD:
+  firebase_admin.initialize_app(credentials.Certificate(get_credentials_json()))
+  db = firestore.client()
+
+
+class Category(enum.Enum):
+  SUMMARIZATION = "summarization"
+  TRANSLATION = "translation"
+
+
+class Rating:
+
+  def __init__(self, model: str, rating: int):
+    self.model = model
+    self.rating = rating
+
+
+def get_ratings(category: Category, source_lang: str | None,
+                target_lang: str | None) -> List[Rating] | None:
+  doc_id = "#".join([category.value] +
+                    [lang for lang in (source_lang, target_lang) if lang])
+  doc_dict = db.collection(RATINGS_COLLECTION).document(doc_id).get().to_dict()
+  if doc_dict is None:
+    return None
+
+  # TODO(#37): Return the timestamp as well.
+  doc_dict.pop("timestamp")
+
+  return [Rating(model, rating) for model, rating in doc_dict.items()]
+
+
+def set_ratings(category: Category, ratings: List[Rating],
+                source_lang: str | None, target_lang: str | None):
+  doc_id = "#".join([category.value] +
+                    [lang for lang in (source_lang, target_lang) if lang])
+  doc_ref = db.collection(RATINGS_COLLECTION).document(doc_id)
+
+  batch = db.batch()
+  for rating in ratings:
+    batch.set(doc_ref, {rating.model: rating.rating}, merge=True)
+  batch.set(doc_ref, {"timestamp": firestore.SERVER_TIMESTAMP}, merge=True)
+  batch.commit()
```

Comment:
Any reason to use batch? I don't think it's necessary because you're modifying a single document.
How about merging the changes?

Reply:
you're right. I'll merge it